### DISCUSSION TITLE
How to generate the openapi.json spec file

### DISCUSSION THREAD

**Discussion by denismp** [OAI/OpenAPI-Specification#3272] [Q&A]
I am completely new to open api.  Is there a tool that one can use to generate this file, perhaps against a RESTFULL API service repo or does it need to be manually edited?  If so, is there an editor that can be used for that purpose?

Can yaml be used instead of json?  I am trying to figure out what I need to feed to the zap_api_scan.py -target argument.

**Comment by MikeRalphson** [comment]
I'm guessing you're using Python? Take a look at https://tools.openapis.org to find tools and frameworks which can generate an OpenAPI definition from existing code, using annotations. You can certainly create one by hand, in various text or GUI editors (such as VsCode extensions, `openapi-gui`, Postman, Swagger Editor/SwaggerHub etc.

YAML can indeed be used in place of JSON (the spec. says so, unfortunately a small amount of tooling is JSON only).

**Comment by handrews** [comment]
Closing as this is a tooling question and an earlier response directed folks to a list of tools.
